groups/script.js

```diff
@@ -58,6 +58,9 @@ groupsData?.forEach((item) => {
   group.appendChild(createdBy);
   group.setAttribute('id', item.roleid);
   group.classList.add('group-role');
+  if (window.location.search.slice(1) === item.rolename) {
```

Comment:
in the line window.location.search.slice(1). The slice() method returns a new string that contains a portion of the original string. In this case, the slice() method is being used to return the second character of the window.location.search string. However, the window.location.search string is a URL query string, and the second character of a URL query string is always the ? character. So, the slice() method will always return an empty string.

To fix the bug, you need to change the line window.location.search.slice(1) to window.location.search.substring(2). The substring() method also returns a new string that contains a portion of the original string, but it starts at the specified index and includes all characters up to the next index. In this case, the substring() method will start at the third character of the window.location.search string and include all characters up to the end of the string.

Here is the corrected code:

if (window.location.search.substring(2) === item.rolename) {
    group.classList.add('active-group');
}

Reply:
Here, window.location.search.slice(1) is returning new string by removing first character from original string.
For example, if URL is https://dev.realdevsquad.com/groups/?group-chess
window.location.search will give ?group-chess
window.location.search.slice(1) will give group-chess.

Please let me know if this is wrong.